### DISCUSSION TITLE
NMF outputs

### DISCUSSION THREAD

**Discussion by rue1996** [BayraktarLab/cell2location#154] [Q&A]
Hi i am trying to understand the NFM outputs. What is the difference between NFM compartments and leiden clusters? Which one of the two is more accurate? Also I am trying to use the NFM outputs for further downstream analysis, I see that the output provides the top two cell types in each factor, I was wondering is it possible to obtain the marker genes of each factor?

**Comment by vitkl** [comment]
Hi @rue1996 

NMF factors and clusters have a different interpretation. See cell2location supplementary information (https://www.nature.com/articles/s41587-021-01139-4#Sec50) page 34 for head to head comparison (screenshot below) and section 4.2 for detailed explanation of how NMF can be interpreted (page 60 in PDF).

<img width="572" alt="Screenshot 2022-05-28 at 00 14 23" src="https://user-images.githubusercontent.com/22567383/170799286-f27d21c8-d5b6-467e-b401-65ed709cf9db.png">
